site/zenodo_rdm/stats/exporters.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# ZenodoRDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""ZenodoRDM stats exporters."""
+
+import json
+from urllib.parse import urlencode, urlsplit, urlunsplit
+
+import requests
+from dateutil.parser import parse as dateutil_parse
+from flask import current_app, url_for
+from invenio_cache import current_cache
+from invenio_pidstore.errors import PIDDeletedError
+from invenio_search import current_search_client
+from invenio_search.utils import build_alias_name
+from opensearch_dsl import Search
+
+from zenodo_rdm.stats.errors import PiwikExportRequestError
+from zenodo_rdm.stats.utils import chunkify, fetch_record
+
+
+class PiwikExporter:
+    """Events exporter."""
+
+    def run(self, start_date=None, end_date=None, update_bookmark=True):
+        """Run export job."""
+        if start_date is None:
+            bookmark = current_cache.get("piwik_export:bookmark")
```

Comment:
note/shelve: we need to remember to "migrate" this bookmark value from the legacy system to production (so that we don't resend statistics from the very beginning).

But we'll also be reminded by the logged warning below :)